File: internal/content/testdata/bench/go/nc-graph-valid-tree.go
func valid_tree(n int, edges [][]int) bool {
	if len(edges) != n-1 {
		return false
	}
	adj := make([][]int, n)
	for _, e := range edges {
		u, v := e[0], e[1]
		adj[u] = append(adj[u], v)
		adj[v] = append(adj[v], u)
	}
	visited := make([]bool, n)
	var dfs func(node int)
	dfs = func(node int) {
		visited[node] = true
		for _, nb := range adj[node] {
			if !visited[nb] {
				dfs(nb)
			}
		}
	}
	dfs(0)
	for _, v := range visited {
		if !v {
			return false
		}
	}
	return true
}

File: internal/content/testdata/bench/java/nc-graph-valid-tree.java
import java.util.*;

class Solution {
    public boolean valid_tree(long n, long[][] edges) {
        if (edges.length != n - 1) return false;
        int N = (int) n;
        List<List<Integer>> adj = new ArrayList<>();
        for (int i = 0; i < N; i++) adj.add(new ArrayList<>());
        for (long[] e : edges) {
            int u = (int) e[0], v = (int) e[1];
            adj.get(u).add(v);
            adj.get(v).add(u);
        }
        boolean[] visited = new boolean[N];
        dfs(0, adj, visited);
        for (boolean v : visited) {
            if (!v) return false;
        }
        return true;
    }

    private void dfs(int node, List<List<Integer>> adj, boolean[] visited) {
        visited[node] = true;
        for (int nb : adj.get(node)) {
            if (!visited[nb]) dfs(nb, adj, visited);
        }
    }
}
